File: docs/snippets/csharp/cache/disk_cache.cs
```csharp title="disk_cache.cs"
using Xberg;
using System;
using System.IO;
using System.Threading.Tasks;

var config = new ExtractionConfig
{
    UseCache = true,
    CacheConfig = new CacheConfig
    {
        CachePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "xberg_cache"),
        MaxCacheSize = 1024 * 1024 * 500, 
        CacheTtlSeconds = 86400 * 7,      
        EnableCompression = true
    }
};

Console.WriteLine("First extraction (will be cached)...");
var result1 = await XbergLib.ExtractAsync("document.pdf", config);
Console.WriteLine($"  - Content length: {result1.Content.Length}");
Console.WriteLine($"  - Cached: {result1.Metadata.WasCached}");

Console.WriteLine("\nSecond extraction (from cache)...");
var result2 = await XbergLib.ExtractAsync("document.pdf", config);
Console.WriteLine($"  - Content length: {result2.Content.Length}");
Console.WriteLine($"  - Cached: {result2.Metadata.WasCached}");

Console.WriteLine($"\nResults are identical: {result1.Content == result2.Content}");

await XbergLib.ClearCacheAsync("document.pdf");
Console.WriteLine("\nCache cleared for document.pdf");

await XbergLib.ClearAllCacheAsync();
Console.WriteLine("All cache cleared");

var cacheStats = await XbergLib.GetCacheStatsAsync();
Console.WriteLine($"\nCache Statistics:");
Console.WriteLine($"  - Total entries: {cacheStats.TotalEntries}");
Console.WriteLine($"  - Cache size: {cacheStats.CacheSizeBytes / 1024 / 1024} MB");
Console.WriteLine($"  - Hit rate: {cacheStats.HitRate:P}");
```

File: docs/snippets/java/cache/DiskCache.java
```java title="DiskCache.java"
import com.xberg.*;
import java.nio.file.Files;
import java.nio.file.Paths;

public final class DiskCache {
  private DiskCache() {}

  public static void main(String[] args) throws Exception {
    String cacheDir = System.getProperty("user.home") + "/.cache/xberg";
    Files.createDirectories(Paths.get(cacheDir));

    CacheConfig cacheConfig = new CacheConfig(
      cacheDir,
      500L * 1024 * 1024,
      7L * 86400,
      true
    );

    ExtractionConfig config = new ExtractionConfig.Builder()
      .useCache(true)
      .cacheConfig(cacheConfig)
      .build();

    Xberg xberg = new Xberg(config);

    System.out.println("First extraction (will be cached)...");
    ExtractionResult result1 = xberg.extract("document.pdf");
    System.out.println("  - Content length: " + result1.content().length());
    System.out.println("  - Cached: " + result1.metadata().wasCached());

    System.out.println("\nSecond extraction (from cache)...");
    ExtractionResult result2 = xberg.extract("document.pdf");
    System.out.println("  - Content length: " + result2.content().length());
    System.out.println("  - Cached: " + result2.metadata().wasCached());

    System.out.println("\nResults are identical: " + result1.content().equals(result2.content()));

    CacheStats cacheStats = xberg.getCacheStats();
    System.out.println("\nCache Statistics:");
    System.out.println("  - Total entries: " + cacheStats.totalEntries());
    System.out.println("  - Cache size: " + String.format("%.1f", cacheStats.cacheSizeBytes() / 1024.0 / 1024.0) + " MB");
    System.out.println("  - Hit rate: " + String.format("%.1f", cacheStats.hitRate() * 100) + "%");
  }
}
```
